### PR TITLE
Export the `SKOLEM` kind

### PR DESCRIPTION
**Pull request by ajreynol**
This distinguishes a new kind of internal skolem `DUMMY_SKOLEM` for all skolems that are not formalized internally. 

We now have the invariant that terms with internal kind `cvc5::internal::Kind::SKOLEM` are (exportable) skolems in our API.

Thus, calls in the SkolemManager e.g. `isSkolemFunction` now are simply a Kind check.

Moreover, `cvc5::internal::Kind::SKOLEM` is now mapped to a new `cvc5::Kind::SKOLEM` kind.

Note that https://github.com/cvc5/cvc5/pull/10559 should go in first, since `INPUT_VARIABLE` is an exception to the invariant introduced here.

FYI @HanielB

### CODE REVIEW

File: src/expr/skolem_manager.cpp [hunk 1]
```diff
@@ -212,21 +212,20 @@ Node SkolemManager::mkSkolemFunctionTyped(SkolemFunId id,
 
 bool SkolemManager::isSkolemFunction(TNode k) const
 {
-  std::map<Node, std::tuple<SkolemFunId, TypeNode, Node>>::const_iterator it =
-      d_skolemFunMap.find(k);
-  return it != d_skolemFunMap.end();
+  return k.getKind()==Kind::SKOLEM;
```
Comment (abdoo8080) [inline src/expr/skolem_manager.cpp:215]: Format

File: src/expr/skolem_manager.cpp [hunk 2]
```diff
@@ -212,21 +212,20 @@ Node SkolemManager::mkSkolemFunctionTyped(SkolemFunId id,
...
-  if (it == d_skolemFunMap.end())
+  if (k.getKind()!=Kind::SKOLEM)
   {
     return false;
   }
+  std::map<Node, std::tuple<SkolemFunId, TypeNode, Node>>::const_iterator it =
+      d_skolemFunMap.find(k);
+  Assert (it != d_skolemFunMap.end());
   id = std::get<0>(it->second);
   cacheVal = std::get<2>(it->second);
```
Comment (abdoo8080) [inline src/expr/skolem_manager.cpp:230]: What's the point of those 2 lines?

File: src/proof/alf/alf_node_converter.cpp [hunk 3]
```diff
@@ -79,7 +79,7 @@ Node AlfNodeConverter::postConvert(Node n)
     return n;
   }
   TypeNode tn = n.getType();
-  if (k == Kind::SKOLEM)
+  if (k == Kind::SKOLEM || k==Kind::DUMMY_SKOLEM)
```
Comment (abdoo8080) [inline src/proof/alf/alf_node_converter.cpp:82]: Format

File: src/expr/skolem_manager.cpp [hunk 4]
```diff
@@ -212,21 +212,20 @@ Node SkolemManager::mkSkolemFunctionTyped(SkolemFunId id,
...
 }
 
 bool SkolemManager::isSkolemFunction(TNode k,
                                      SkolemFunId& id,
                                      Node& cacheVal) const
 {
-  std::map<Node, std::tuple<SkolemFunId, TypeNode, Node>>::const_iterator it =
-      d_skolemFunMap.find(k);
-  if (it == d_skolemFunMap.end())
+  if (k.getKind()!=Kind::SKOLEM)
```
Comment (abdoo8080) [inline src/expr/skolem_manager.cpp:222]: Format

File: src/theory/sort_inference.cpp [hunk 5]
```diff
@@ -784,7 +784,7 @@ Node SortInference::simplifyNode(
       if( it!=var_bound.end() ){
         ret = it->second;
       }
-      else if (n.getKind() == Kind::VARIABLE || n.getKind() == Kind::SKOLEM)
+      else if (n.isVar() && n.getKind()!=Kind::BOUND_VARIABLE)
```
Comment (abdoo8080) [inline src/theory/sort_inference.cpp:787]: Format

File: src/theory/builtin/kinds [hunk 6]
```diff
@@ -296,7 +296,8 @@ operator EQUAL 2 "equality (two parameters only, sorts must match)"
 operator DISTINCT 2: "disequality (N-ary, sorts must match)"
 variable VARIABLE "a variable (not permitted in bindings)"
 variable BOUND_VARIABLE "a bound variable (permitted in bindings and the associated lambda and quantifier bodies only)"
-variable SKOLEM "a Skolem variable (internal only)"
+variable SKOLEM "a Skolem variable"
+variable DUMMY_SKOLEM "a Skolem variable (internal only)"
```
Comment (abdoo8080) [inline src/theory/builtin/kinds:300]: Why not add `DUMMY_SKOLEM` as an `InternalSkolemId`?

File: src/expr/skolem_manager.cpp (hunk 2 above)
Comment (ajreynol) [reply src/expr/skolem_manager.cpp:230]: This retrieves the information about the skolem (this method is the overload where the caller wants to get the definition of the skolem in addition to knowing whether it is a skolem).

File: src/theory/builtin/kinds (hunk 6 above)
Comment (ajreynol) [reply src/theory/builtin/kinds:300]: Skolems constructed via `InternalSkolemId` are constructed based on a `cacheVal` and are such that the `SkolemManager` caches them, they have `SKOLEM` kind but skolem id `INTERNAL`.  This makes it simple that the set of skolems cached in `SkolemManager` are exactly those with kind `SKOLEM`.

Dummy skolems are even less restricted: they are just simply fresh variables constructed without caching.  Making them have a different kind simplifies a lot of the logic.  

In the longer term, we should get rid of all calls to `mkDummySkolem`.

**Pull request merged**